core/src/main/java/org/apache/iceberg/hadoop/HadoopFileIO.java

```diff
@@ -89,4 +97,62 @@ public Configuration getConf() {
   public void serializeConfWith(Function<Configuration, SerializableSupplier<Configuration>> confSerializer) {
     this.hadoopConf = confSerializer.apply(getConf());
   }
+
+  @Override
+  public Stream<FileInfo> listPrefix(String prefix) {
+    Path prefixToList = new Path(prefix);
+    FileSystem fs = Util.getFs(prefixToList, hadoopConf.get());
+
+    try {
+      return Streams.stream(new AdaptingIterator<>(fs.listFiles(prefixToList, true)))
```

Comment:
Minor: when passing boolean arguments, we prefer to add an inline comment to explain them: `fs.listFiles(prefixToList, true /* recursive *);`